include/quicly/loss.h

```diff
@@ -188,18 +188,27 @@ inline void quicly_loss_update_alarm(quicly_loss_t *r, int64_t now, int64_t last
     }
 }
 
-inline void quicly_loss_on_ack_received(quicly_loss_t *r, uint64_t largest_acked, uint32_t latest_rtt, uint32_t ack_delay,
-                                        int is_ack_only)
+inline void quicly_loss_on_ack_received(quicly_loss_t *r, uint64_t largest_newly_acked, int64_t now, int64_t sent_at,
+                                        uint32_t ack_delay, size_t bytes_acked)
 {
-    if (r->largest_acked_packet < largest_acked)
-        r->largest_acked_packet = largest_acked;
-    if (latest_rtt != UINT32_MAX)
-        quicly_rtt_update(&r->rtt, latest_rtt, ack_delay);
-}
-
-inline void quicly_loss_on_newly_acked(quicly_loss_t *r)
-{
-    r->pto_count = 0;
+    if (largest_newly_acked != UINT64_MAX)
+        r->pto_count = 0;
+
+    /* Only use RTT samples for new largest acked */
+    if (r->largest_acked_packet >= largest_newly_acked)
+        return;
```

Comment:
I am not sure if this escape condition is correct.

When there is no newly acked packet, `largest_newly_acked` is set to UINT64_MAX. In such case, the `if` would always fail (unless you set `r->largest_acked_packet` to UINT64_MAX, which you should not, because we'd never be emitting a PN that's so large).

Reply:
Good catch! I've fixed this now.